tests/parametric/test_otel_span_methods.py

```diff
@@ -396,6 +398,212 @@ def test_otel_set_attributes_separately(self, test_agent, test_library):
             assert span["name"] == "kafka.receive"
             assert span["resource"] == "operation"
 
+    @missing_feature(context.library < "java@1.26.0", reason="Implemented in 1.26.0")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_started_with_link_from_another_span(self, test_agent, test_library):
+        """Test adding a span link created from another span.
+        This tests the functionality of "create a direct link between two spans
+        given two valid span (or SpanContext) objects" as specified in the RFC.
+        """
+        with test_library:
+            with test_library.otel_start_span("root") as parent:
+                parent.end_span()
+                with test_library.otel_start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
+                ) as child:
+                    child.end_span()
```

Comment:
nit: The span should ended after the `with` block terminates
```suggestion
                    pass
```

Reply:
I just tried and I only have a trace with 1 span instead of 2